src/components/modal/lagrede-filter/lagre-filter-modal.tsx

```diff
@@ -0,0 +1,225 @@
+                    </>
+                    : valgtLagretFilter
+                        ? <>
+                            <Normaltekst>Det finnes allerede et lagret
+                                filter <b>"{valgtLagretFilter!.filterNavn}"</b> med
+                                denne filterkombinasjonen. Oppdater navnet ved å klikke på knappen under.
+                            </Normaltekst>
+                            <br/>
+                            {oppdaterFilterKnapp()}
+                        </>
+                        : sisteValgteLagredeFilter
+                            ? <>
+                                {lagreNyttFilterKnapp()}
+                                <Normaltekst>Oppdater <b>"{lagretFilterNavn(sisteValgteLagredeFilter)}"</b> ved å klikke
+                                    på
+                                    knappen under.</Normaltekst>
+                                {oppdaterFilterKnapp()}
+                            </>
+                            : <>
+                                {lagreNyttFilterKnapp()}
+                            </>
+                }
+            </div>
+        )
+    }
+
+    const modalTittel = () => {
+        if (valgtVisningstype === Visningstype.LAGRE_NYTT) {
+            return "Lagre nytt filter"
+        } else if (valgtVisningstype === Visningstype.OPPDATER) {
+            return "Endre filter"
+        }
+        return "Lagre filter"
+    }
+
+    useEffect(() => {
+        if (saveRequestSent && tillatteHandlinger.includes(lagretFilterHandlingsType!)) {
+            if (lagretFilterStatus === STATUS.PENDING) {
+            } else if (lagretFilterStatus === STATUS.ERROR) {
+                setErrorModalErApen(true)
+                setSaveRequestSent(false)
+            } else if (lagretFilterStatus === STATUS.OK) {
+                setSaveRequestSent(false)
+                props.onRequestClose();
+                setValgtVisningstype(props.velgVisningstype)
+            }
+        }
+    }, [lagretFilterStatus, lagretFilterHandlingsType, saveRequestSent, tillatteHandlinger, props])
+
+    return (
+        <>
+            <Modal
+                className="lagret-filter-meny-modal"
+                contentLabel="Lagre filter meny modal"
+                isOpen={props.isOpen}
+                onRequestClose={lukkModal}
+                tittel={modalTittel()}
+            >
+                <div className="modal-visningstype">
+
+                    {valgtVisningstype === Visningstype.MENY &&
+                    <Meny/>
+                    }
+
+                    {valgtVisningstype === Visningstype.LAGRE_NYTT &&
+                    <LagreNytt
+                        feilValidering={feilValidering}
+                        feilmelding={lagretFilterValidering}
+                        lukkModal={lukkModal}
+                        saveRequestSent={setSaveRequestSent}
+                        filterNavn={filterNavn}
```

Comment:
hvorför blir filternavn sent når man vill lagre som nytt filter? borde inte det hellre skickas till oppdatere filter?

Reply:
I agree it could be better, so the logic is changed, please check.